model/modelprocessor/errormessage.go

```diff
@@ -20,25 +20,25 @@ package modelprocessor
 import (
 	"context"
 
-	"github.com/elastic/apm-data/model"
+	"github.com/elastic/apm-data/model/modelpb"
 )
 
 // SetErrorMessage is a model.BatchProcessor that sets the APMEvent.Message
 // field for error events.
 type SetErrorMessage struct{}
 
 // ProcessBatch sets the message for errors.
-func (s SetErrorMessage) ProcessBatch(ctx context.Context, b *model.Batch) error {
+func (s SetErrorMessage) ProcessBatch(ctx context.Context, b *modelpb.Batch) error {
 	for i := range *b {
-		event := &(*b)[i]
+		event := (*b)[i]
 		if event.Error != nil {
 			event.Message = s.setErrorMessage(event)
 		}
 	}
 	return nil
 }
 
-func (s SetErrorMessage) setErrorMessage(event *model.APMEvent) string {
+func (s SetErrorMessage) setErrorMessage(event *modelpb.APMEvent) string {
```

Comment:
This function can be simplified using getters:

```go
func (s SetErrorMessage) setErrorMessage(event *modelpb.APMEvent) string {
	if msg := event.GetError().GetLog().GetMessage(); msg != "" {
		return msg
	}
	if msg := event.GetError().GetException().GetMessage(); msg != "" {
		return msg
	}
	return ""
}
```

Reply:
Good catch!